addons/mail/static/src/components/attachment_link_preview/tests/attachment_link_preview_tests.js

```diff
@@ -0,0 +1,154 @@
+/** @odoo-module **/
+
+import { link } from '@mail/model/model_field_command';
+import {
+    afterEach,
+    afterNextRender,
+    beforeEach,
+    createRootMessagingComponent,
+    start,
+} from '@mail/utils/test_utils';
+
+QUnit.module('mail', {}, function () {
+QUnit.module('components', {}, function () {
+QUnit.module('attachment_link_preview', {}, function () {
+QUnit.module('attachment_link_preview_tests.js', {
+    beforeEach() {
+        beforeEach(this);
+
+        this.createMessageComponent = async (message, otherProps) => {
+            const props = Object.assign({ messageLocalId: message.localId }, otherProps);
+            await createRootMessagingComponent(this, "Message", {
+                props,
+                target: this.widget.el,
+            });
+        };
+
+        this.start = async params => {
+            const { env, widget } = await start(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        afterEach(this);
+    },
+});
+
+QUnit.test('Link preview card is not viewable', async function(assert) {
+    assert.expect(1);
+
+    await this.start();
+    const attachment = this.messaging.models['mail.attachment'].create({
+        description: 'test description',
+        filename: "https://tenor.com/view/gato-gif-18532922",
+        id: 750,
+        mimetype: 'application/o-linkpreview',
+        name: "https://tenor.com/view/gato-gif-18532922",
+        url: "https://tenor.com/view/gato-gif-18532922",
+    });
+    const message = this.messaging.models['mail.message'].create({
+        attachments: link(attachment),
+        author: link(this.messaging.currentPartner),
+        body: "<p>Test</p>",
+        id: 100,
+    });
+    await this.createMessageComponent(message);
+
+    assert.containsOnce(
+        document.body,
+        '.o_AttachmentLinkPreview',
+        "Attachment is a link preview"
+    );
+});
+
+QUnit.test('simplest layout', async function (assert) {
+    assert.expect(4);
+
+    await this.start();
+    const attachment = this.messaging.models['mail.attachment'].create({
+        description: 'test description',
+        filename: "https://tenor.com/view/gato-gif-18532922",
+        id: 750,
+        mimetype: 'application/o-linkpreview',
+        name: "https://tenor.com/view/gato-gif-18532922",
+        url: "https://tenor.com/view/gato-gif-18532922",
+    });
+    const message = this.messaging.models['mail.message'].create({
+        attachments: link(attachment),
+        author: link(this.messaging.currentPartner),
+        body: "<p>Test</p>",
+        id: 100,
+    });
```

Comment:
I'm skipping over this file for now but the same comments as other test file applies.